docs/book/src/tasks/change-machine-template.md

```diff
@@ -0,0 +1,35 @@
+# Changing Infrastructure Machine Templates
+
+Several different components of Cluster API leverage _infrastructure machine templates_, 
+including `KubeadmControlPlane`, `MachineDeployment`, and `MachineSet`. These 
+`MachineTemplate` resources are **immutable** and are not intended to be modified 
```

Comment:
xref https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/1711. I know you mention at the bottom that providers may decide to support some field changes, but I wonder if we should highlight this use case here, or soften/clarify the **immutable** bit here?

Reply:
+1 I'd say that they _should_ be immutable, unless the documentation on the infrastructure provider states otherwise for certain fields

cc @rudoi @sethp-nr

Reply:
Addressed via af55bb4.